Add run-image-mirrors to config subcommand

dfreilich opened the pull request:
Signed-off-by: David Freilich <dfreilich@vmware.com>

## Summary
<!-- Provide a high-level summary of the change. -->
This adds a new subcommand `run-image-mirrors` to the `pack config` subcommand, with 3 further commands:
* `pack config run-image-mirrors add`
* `pack config run-image-mirrors remove`
* `pack config run-image-mirrors list`

This maps and hides `pack set-run-image-mirrors`

I changed the behavior for `pack set-run-image-mirrors`/`pack config run-image-mirrors add`. Formerly, it overwrote the previously set values for the `runImage` you want to save. Now, we append, dedup and sort the mirrors. (I thought that it made more sense, but not a hill I will 👊  on)

I debated adding explicit acceptance tests for the methods, but since they are self-contained, I felt that they were well covered through the unit tests. 

## Output
<!-- If applicable, please provide examples of the output changes. -->

#### Before

#### After

## Documentation
<!-- If this change should be documented, please create an issue or PR on https://github.com/buildpacks/docs and link below. -->
<!-- NOTE: This can be added (by editing the issue) after the PR is opened. -->

- Should this change be documented?
    - [ ] Yes, see #___
    - [ ] No

## Related
<!-- If this PR addresses an issue, please provide issue number below. -->

Resolves #922 
Relates to #597

jromero commented:
This is awesome! I like the overall change from "set" to "add". It _feels_ natural.

There are some issue that we may want to address before merging it in:

## `... run-image-mirrors add`

### ⁉️ `--help` doesn't show arg requirement

#### Output

```
❯ pack-dev config run-image-mirrors add --help
Set mirrors to other repositories for a given run image

Usage:
  pack config run-image-mirrors add [flags]

Examples:
pack set-run-image-mirrors cnbs/sample-stack-run:bionic --mirror index.docker.io/cnbs/sample-stack-run:bionic
```

### ⁉️ `.. --help` returns example for deprecated

#### Output

```
❯ pack-dev config run-image-mirrors add --help
Set mirrors to other repositories for a given run image

Usage:
  pack config run-image-mirrors add [flags]

Examples:
pack set-run-image-mirrors cnbs/sample-stack-run:bionic --mirror index.docker.io/cnbs/sample-stack-run:bionic
```

## `... run-image-mirrors list`

### ⁉️ "Run Image Mirrors" is displayed multiple times

After adding some run-image-mirrors I see that "Run Image Mirrors:" is displayed multiple times.

#### Output

```
❯ pack-dev config run-image-mirrors
Run Image Mirrors:
Run Image Mirrors:
  cnbs/sample-stack-run:bionic:
    index.docker.io/cnbs/sample-stack-run:bionic
    index.docker.io/cnbs/sample-stack-run:bionic1
Run Image Mirrors:
  cnbs/sample-stack-run:bionik:
    index.docker.io/cnbs/sample-stack-run:bionik
```

## `... run-image-mirrors remove`

### ⁉️ `--help` doesn't show arg requirement

### Output

```
❯ pack-dev config run-image-mirrors remove --help
Remove a run-image-mirrors

Usage:
  pack config run-image-mirrors remove [flags]

Flags:
  -h, --help             help for remove
  -m, --mirror strings   Run image mirror
                         Repeat for each mirror in order,
                           or supply once by comma-separated list

Global Flags:
      --no-color     Disable color output
  -q, --quiet        Show less output
      --timestamps   Enable timestamps in output
  -v, --verbose      Show more output
```

```
❯ pack-dev config run-image-mirrors remove --mirror index.docker.io/cnbs/sample-stack-run:bionik
Error: accepts 1 arg(s), received 0
Usage:
  pack config run-image-mirrors remove [flags]

Flags:
  -h, --help             help for remove
  -m, --mirror strings   Run image mirror
                         Repeat for each mirror in order,
                           or supply once by comma-separated list

Global Flags:
      --no-color     Disable color output
  -q, --quiet        Show less output
      --timestamps   Enable timestamps in output
  -v, --verbose      Show more output
```

### ⁉️ doesn't provide output

#### Output

```
❯ pack-dev config run-image-mirrors remove cnbs/sample-stack-run:bionik --mirror index.docker.io/cnbs/sample-stack-run:bionik

```

#### Expected

Some output that told me it was successful in removing the mirror

### ⁉️ leaves empty image entry in list output

#### Output

```
❯ pack-dev config run-image-mirrors remove cnbs/sample-stack-run:bionik --mirror index.docker.io/cnbs/sample-stack-run:bionik

❯ pack-dev config run-image-mirrors list
Run Image Mirrors:
Run Image Mirrors:
  cnbs/sample-stack-run:bionic:
    index.docker.io/cnbs/sample-stack-run:bionic
    index.docker.io/cnbs/sample-stack-run:bionic1
Run Image Mirrors:
  cnbs/sample-stack-run:bionik:
```

#### Expected

I expected `cnbs/sample-stack-run:bionik` to no longer be listed since it has no mirrors.

jromero commented:
One more I just found:

### ⁉️ `config --help` does not provide description for `run-image-mirrors`

#### Output

```
pack-dev config --help
Interact with Pack's configuration

Usage:
  pack config [command]

Available Commands:
  trusted-builders  Interact with trusted builders
  run-image-mirrors 
```

dfreilich commented:
@jromero Thanks for all the feedback! I fixed most of it, let me know what you think:

### `-h` flags
![image](https://user-images.githubusercontent.com/7035673/101341261-7d5f3a80-3889-11eb-8136-68e73dff2881.png)

![image](https://user-images.githubusercontent.com/7035673/101341278-8223ee80-3889-11eb-9ae5-a36cddb5baab.png)

### commands
![image](https://user-images.githubusercontent.com/7035673/101341418-bac3c800-3889-11eb-9675-5c398e13e261.png)

(I changed the return string when you don't provide any mirrors for run image)
